crates/bevy_ecs/src/system/system.rs

```diff
@@ -94,6 +94,32 @@ pub trait System: Send + Sync + 'static {
     /// of this system into the world's command buffer.
     fn queue_deferred(&mut self, world: DeferredWorld);
 
+    /// Validates that all systems parameters can be acquired.
+    /// If not, the system should not be ran.
+    ///
+    /// # Safety
+    ///
+    /// - The caller must ensure that `world` has permission to read any world data
+    ///   registered in [`Self::archetype_component_access`]. There must be no conflicting
```

Comment:
Why does this mention `update_archetype_component_access` while the other `validate_param`s mention `init_state`?

Reply:
This was copy-paste from `get_param`/`run` methods, this means that they are also inconsistent with each other.
Agreed it should refer to `update_archetype_component_access` and some reference to `new_archetype` instead.

Reply:
Addressed, let me know if that's sufficient.
If not I'm afraid we'll need a follow up PR since Alice wants this merged in the next train

Reply:
I guess it's good enough now, I'm ok with merging it and maybe improve the wording in a follow up PR